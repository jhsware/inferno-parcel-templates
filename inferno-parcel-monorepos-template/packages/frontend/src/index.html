<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>Inferno Monorepo Template</title>
    <script type="module" src="index.ts"></script>
  </head>
  <body>
    <div id="app"></div>
  </body>
</html>