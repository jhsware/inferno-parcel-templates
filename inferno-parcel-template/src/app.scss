article {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  min-height: 30vh;
}