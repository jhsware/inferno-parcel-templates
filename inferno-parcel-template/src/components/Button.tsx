type TButton = {
  className?: string | undefined; 
  submit?: boolean;
  children: any;
  onClick?(e: MouseEvent): void;
}

export function Button({className = undefined, submit = false, children = undefined, onClick}: TButton) {  
  const props = {
    className: className ? `Button ${className}` : 'Button',
    onClick,
    submit,
  }

  return (
    <button {...props}>
      {children}
    </button>
  )
}
