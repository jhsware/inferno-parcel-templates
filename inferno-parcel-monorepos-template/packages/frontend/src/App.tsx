import { Button } from 'components';
import './app.scss'

/**
 * The Application
 */
type TApp = {
  title: string,
  children: any
}
export function App({ title, children }: TApp) {
  return (
    <article>
      <h1>{title}</h1>
      {children}
    </article>
  )
};

export function appFactory () {

  return (
      <App title="Inferno Monorepo Template">
        <Button onClick={(e) => alert("This works!")}>Click me!</Button>
      </App>
  )
}
