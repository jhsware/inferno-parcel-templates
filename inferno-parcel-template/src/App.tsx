import { Button } from './components/Button';
import './app.scss'

/**
 * The Application
 */
type TApp = {
  title: string,
  children: any
}
export function App({ title, children }: TApp) {
  return (
    <article>
      <h1>{title}</h1>
      {children}
    </article>
  )
};

export function appFactory () {

  return (
      <App title="Inferno Template">
        <Button onClick={(e) => alert("This works!")}>Click me!</Button>
      </App>
  )
}
