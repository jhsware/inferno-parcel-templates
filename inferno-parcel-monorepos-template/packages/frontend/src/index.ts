import { render } from 'inferno';
import { appFactory } from './App';

render(appFactory(), document.getElementById('app'))
