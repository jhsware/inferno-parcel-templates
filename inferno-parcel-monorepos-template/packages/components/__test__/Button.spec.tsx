import { render } from 'inferno';
import { describe, expect, it, beforeEach } from "@jest/globals";

import { Button } from "../src/Button";

describe('Button', () => {
  let node;
  beforeEach(() => {
    node = document.createElement('div');
  })

  it('can be rendered', () => {
    render(<Button>Click</Button>, node);
    expect(node).toMatchSnapshot();
  });
});