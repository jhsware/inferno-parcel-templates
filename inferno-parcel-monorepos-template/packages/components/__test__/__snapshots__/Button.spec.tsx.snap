// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can be rendered 1`] = `
<div>
  <button
    class="Button"
    submit="false"
  >
    Click
  </button>
</div>
`;
